### ISSUE TITLE
Automatically pin all files to local IPFS node

### ISSUE THREAD

**Issue by bmann** [fission-suite/fission#329]
From the forum: https://talk.fission.codes/t/automatically-pin-locally/640

> As long as IPFS is running on my machine I do not have to wait for huge uploads to complete before I can share a video for example, or any of my stuff.
>
>Hopefully fission drive could arrange to pin everything I have locally, and tell me when it is all done. To know that even if I turn my machine off it will be available.

> I would like to tell fission.drive, here is a cid for something I just pinned on my machine,
please make sure that it is pinned on fission drive too.

Likely this can be a very small change, where as well as sending the hash to pin remotely, we also pin it locally.

**Comment by expede** [comment]
I believe this is how it works today. @dholms can you investigate?

**Comment by dholms** [comment]
Yeah I'm nearly positive that IPFS automatically pins things when you add them locally. But I'll double check this

**Comment by dholms** [comment]
oops wrong button, reopening

**Comment by bmann** [comment]
This is low priority.

The more interesting question is “what is the best way to make sure that pinning files locally make Fission perform well in a local browser without a plugin”

Which is a larger challenge.

**Comment by dholms** [comment]
Ran a test: `ipfs add` definitely pins that content :+1: 
```
ipfs add ./pic.jpg
ipfs pin ls | grep ${cid}
```

> The more interesting question is “what is the best way to make sure that pinning files locally make Fission perform well in a local browser without a plugin”
hmm yeah this is a bit trickier. You need to get the in browser node to connect to your local node. But most people aren't going to be running their local node with websockets, much less with a proxy which provides SSL. 

We could solve that first part with some quick edits to their IPFS config during `setup`. I wonder if a browser node can connect to a non-secure websockets node on the same machine through a relay hop :thinking:

**Comment by expede** [comment]
> connect to a non-secure websockets node on the same machine through a relay hop

Probably not without running some extra software (e.g. our CLI). We also need to know the PK for the IPFS node. We could actually maybe _do that_ on setup, but that has some not great security implications.

**Comment by bmann** [comment]
I've renamed this, unassigned Daniel, and put it in the Icebox
